model/task_history.go

```diff
@@ -359,46 +359,44 @@ func (self *taskHistoryIterator) GetDistinctTestNames(ctx context.Context, numCo
 	}
 	defer session.Close()
 
-	pipeline := mdb.C(task.Collection).Pipe(
-		[]bson.M{
-			{
-				"$match": bson.M{
-					task.ProjectKey:     self.ProjectName,
-					task.RequesterKey:   evergreen.RepotrackerVersionRequester,
-					task.DisplayNameKey: self.TaskName,
-				},
-			},
-			{"$sort": bson.D{{Key: task.CreateTimeKey, Value: -1}}},
-			{"$limit": numCommits},
-			{"$lookup": bson.M{
-				"from":         testresult.Collection,
-				"localField":   task.IdKey,
-				"foreignField": testresult.TaskIDKey,
-				"as":           testResultsKey},
+	pipeline := []bson.M{
+		{
+			"$match": bson.M{
+				task.ProjectKey:     self.ProjectName,
+				task.RequesterKey:   evergreen.RepotrackerVersionRequester,
+				task.DisplayNameKey: self.TaskName,
 			},
-			{"$project": bson.M{
-				testResultsKey: bson.M{
-					"$filter": bson.M{
-						// Filter off non-matching executions. This should be replaced once
-						// multi-key $lookups are supported in 3.6
-						"input": "$" + testResultsKey,
-						"as":    "tr",
-						"cond": bson.M{
-							"$eq": []string{"$$tr.task_execution", "$execution"}},
-					},
-				},
-				task.IdKey:                 1,
-				task.TestResultTestFileKey: 1,
-			}},
-			{"$unwind": fmt.Sprintf("$%v", testResultsKey)},
-			{"$group": bson.M{"_id": fmt.Sprintf("$%v.%v", testResultsKey, task.TestResultTestFileKey)}},
 		},
-	)
-	pipeline.MaxTime(taskHistoryMaxTime)
+		{"$sort": bson.D{{Key: task.CreateTimeKey, Value: -1}}},
+		{"$limit": numCommits},
+		{"$lookup": bson.M{
+			"from":         testresult.Collection,
+			"localField":   task.IdKey,
+			"foreignField": testresult.TaskIDKey,
+			"as":           testResultsKey},
+		},
+		{"$project": bson.M{
+			testResultsKey: bson.M{
+				"$filter": bson.M{
+					// Filter off non-matching executions. This should be replaced once
+					// multi-key $lookups are supported in 3.6
+					"input": "$" + testResultsKey,
+					"as":    "tr",
+					"cond": bson.M{
+						"$eq": []string{"$$tr.task_execution", "$execution"}},
+				},
+			},
+			task.IdKey:                 1,
+			task.TestResultTestFileKey: 1,
+		}},
+		{"$unwind": fmt.Sprintf("$%v", testResultsKey)},
+		{"$group": bson.M{"_id": fmt.Sprintf("$%v.%v", testResultsKey, task.TestResultTestFileKey)}},
+	}
 
+	agg := db.ConstructPipeWithMaxTime(mdb, task.Collection, pipeline, taskHistoryMaxTime)
```

Comment:
I think here it's okay to not change the function since we use .Iter instead of .All (I'm not sure how long this will exist anyway)